ext/standard/charrepr.h

```diff
@@ -0,0 +1,284 @@
+/*
+  +----------------------------------------------------------------------+
+  | Copyright (c) The PHP Group                                          |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 3.01 of the PHP license,      |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | https://www.php.net/license/3_01.txt                                 |
+  | If you did not receive a copy of the PHP license and are unable to   |
+  | obtain it through the world-wide-web, please send a note to          |
+  | license@php.net so we can mail you a copy immediately.               |
+  +----------------------------------------------------------------------+
+  | Author: Yihang Wang <wangyihanger@gmail.com>                         |
+  +----------------------------------------------------------------------+
+*/
+
+#ifndef STRING_H
+#define STRING_H
+
+typedef struct {
+    const char *repr; 
+    const unsigned char len;
+} char_repr_t;
+
+static const char_repr_t char_reprs[256] = {
```

Comment:
What is the reason to separate this array into a header?
It's going to be duplicated every time the header is included.

Reply:
The reason I separated this array into a header file was primarily for **modularity** and **reusability**. 

The current file (`string.c`) is already quite long, and combining these contents with the C file might make it harder to read and maintain. 

However, if you believe this separation is unnecessary, I can certainly move this part back into `string.c`.

Reply:
Was this code copied from somewhere? There's no header/license that would indicate so.